File: 004/main.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function (nums1, nums2) {
  let s = nums1.concat(nums2).sort((a, b) => {
    return a - b
  })
  let len = s.length
  if (len % 2 === 0) {
    return (s[len / 2 - 1] + s[len / 2]) / 2
  } else {
    return s[Math.floor(len / 2)]
  }
}

// console.log(findMedianSortedArrays([1, 2], [3, 4]))
// console.log(findMedianSortedArrays([1, 3], [2]))
console.log(findMedianSortedArrays([1], [2, 3, 4, 5, 6, 7, 8, 9, 10]))

File: 004/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int> res;
        double ans;
        for (int i = 0; i < nums1.size(); i++) {
            res.push_back(nums1[i]);
        }
        for (int i = 0; i < nums2.size(); i++) {
            res.push_back(nums2[i]);
        }
        sort(res.begin(), res.end());
        int len = res.size();
        if (len % 2 == 1) ans = res[len / 2];
        else ans = (res[len / 2 - 1] + res[len / 2]) * 1.0 / 2;
        return ans;
    }
};

int main()
{
    Solution sol;
    vector<int> nums1, nums2;
    nums1.push_back(1);
    nums1.push_back(3);
    nums2.push_back(2);
    nums2.push_back(4);
    double ans = sol.findMedianSortedArrays(nums1, nums2);
    cout << ans << endl;
    return 0;
}
